lib/pure/uri.nim

```diff
@@ -289,6 +296,16 @@ func parseUri*(uri: string, result: var Uri) =
     assert res.scheme == "https"
     assert res.hostname == "nim-lang.org"
     assert res.path == "/docs/manual.html"
+
+    parseUri("https://nim-lang\n.org\t/docs\nalert('msg\r\n')/?query\n=\tvalue#frag\nment", res)
+    assert res.scheme == "https"
+    assert res.hostname == "nim-lang.org"
+    assert res.path == "/docs/alert('msg')/"
+    assert res.query == "query=value"
+    assert res.anchor == "fragment"
+
```

Comment:
I'm of the opinion that we should raise by default instead of silently removing \r\n\t, see also https://bugs.python.org/msg392995 (or whole thread in https://bugs.python.org/issue43882)

how about adding an optional param `strict = true` ?

Reply:
@timotheecour Yeah, that makes sense. I think it's reasonable to do it as you're suggesting, defaulting to strict and allowing user to handle exception themselves.

The way I understand it, it's also what the WHATWG URL standard suggests:

> 2. If input contains any ASCII tab or newline, validation error.
> 3. Remove all ASCII tab or newline from input.

---
>  A validation error does not mean that the parser terminates. Termination of a parser is always stated explicitly, e.g., through a return statement.
> It is useful to signal validation errors as error-handling can be non-intuitive, legacy user agents might not implement correct error-handling, and the intent of what is written might be unclear to other developers.